### PR TITLE
Feature/defer client evaluation

### PR DESCRIPTION
**Pull request by luke-hill**
I re-opened this because I didn't have the uplink to the previous one and this was easier. So ignore the previous PR

cc/ @hippyhorse

### CODE REVIEW

File: lib/mailosaur.rb [hunk 1]
```diff
@@ -38,37 +38,45 @@ module Models
...
-    def initialize(api_key, base_url)
-      conn = Faraday.new(base_url || 'https://mailosaur.com/', {
+    private
+
+    def connection
+      Faraday.new(@base_url, {
         headers: {
           content_type: 'application/json; charset=utf-8',
-          user_agent: 'mailosaur-ruby/' + Mailosaur::VERSION
+          user_agent: 'mailosaur-ruby/5.0.0'
```
Comment (hippyhorse) [inline lib/mailosaur.rb]: Just need this line reverted please 👍

Comment (luke-hill) [reply lib/mailosaur.rb]: :man_shrugging: I've no idea why it's there, but sure thing

**Comment by hippyhorse** [comment]
Merged, will trigger new release once build succeeds

**Comment by luke-hill** [comment]
For ref this should definitely be a point release x.(+1).0 - I'm not sure on the internal logic of your versioning / if you use semver.

The logic under the hood works slightly differently now and whilst it is entirely an optimisation, you can now access some of the objects in completely different ways / not access some of them at all. Previously you could only access them one way.

**Pull request merged**